interpreter/text/parser.mly

```diff
@@ -488,6 +492,9 @@ limits :
 typeuse :
   | LPAR TYPE idx RPAR { fun c -> $3 c type_ }
 
+heaptypeuse :
+  | LPAR EXACT typeuse RPAR { fun c -> UseHT (Exact, Idx ($3 c).it) }
+  | typeuse { fun c -> UseHT (Inexact, Idx ($1 c).it) }
```

Comment:
If the binary format syntactically allows arbitrary heaptypes, then the text format must, too, to stay equi-expressive. In particular, we should have tests for the invalid cases.